src/NLog/LayoutRenderers/PerformanceCounterLayoutRenderer.cs

```diff
@@ -96,24 +116,67 @@ protected override void InitializeLayoutRenderer()
         {
             base.InitializeLayoutRenderer();
 
-            _prevSample = CounterSample.Empty;
-            _nextSample = CounterSample.Empty;
+            if (_instance == null && string.Equals(Category, "Process", StringComparison.OrdinalIgnoreCase))
+            {
+                _instance = GetCurrentProcessInstanceName(Category) ?? string.Empty;
+            }
+
+            LookupPerformanceCounter(LogEventInfo.CreateNullEvent());
+        }
+
+        /// <inheritdoc />
+        protected override void CloseLayoutRenderer()
+        {
+            base.CloseLayoutRenderer();
+            _fixedPerformanceCounter?.Close();
+            _fixedPerformanceCounter = null;
+            _performanceCounter?.Close();
+            _performanceCounter = null;
+        }
+
+        /// <inheritdoc />
+        protected override void Append(StringBuilder builder, LogEventInfo logEvent)
+        {
+            var performanceCounter = LookupPerformanceCounter(logEvent);
+            var formatProvider = GetFormatProvider(logEvent, Culture);
+            builder.Append(performanceCounter.GetValue().ToString(Format, formatProvider));
```

Comment:
Couldn't .GetValue return `null`?

Reply:
`GetValue()` returns float, so it cannot be null.

Reply:
:+1: